src/snowflake/snowpark/_internal/analyzer/schema_utils.py

```diff
@@ -85,20 +89,50 @@ def analyze_attributes(
     return session._get_result_attributes(sql)
 
 
-def convert_result_meta_to_attribute(meta: List[ResultMetadata]) -> List[Attribute]:
+def convert_result_meta_to_attribute(
+    meta: Union[List[ResultMetadata], List["ResultMetadataV2"]],  # pyright: ignore
+) -> List[Attribute]:
+    # ResultMetadataV2 may not currently be a type, depending on the connector
+    # version, so the argument types are pyright ignored
+
     attributes = []
-    for column_name, type_value, _, internal_size, precision, scale, nullable in meta:
-        quoted_name = quote_name_without_upper_casing(column_name)
+    for column_metadata in meta:
+        quoted_name = quote_name_without_upper_casing(column_metadata.name)
         attributes.append(
             Attribute(
                 quoted_name,
-                convert_sf_to_sp_type(
-                    FIELD_ID_TO_NAME[type_value],
-                    precision or 0,
-                    scale or 0,
-                    internal_size or 0,
-                ),
-                nullable,
+                convert_metadata_to_sp_type(column_metadata),
+                column_metadata.is_nullable,
             )
         )
     return attributes
+
+
+def get_new_description_if_exists(
```

Comment:
Nit: The `if_exists` naming might be a bit confusing and could suggest that `None` might be returned here or the query may not be run if the new describe doesn't exist below. I would suggest dropping it and communicating the fallback behavior in the docstring.